recipes/openai_textcat.py

```diff
@@ -0,0 +1,287 @@
+        "config": {
+            "labels": openai.labels,
+            "batch_size": batch_size,
+            "exclude_by": "input",
+            "choice_style": "single" if exclusive_classes else "multiple",
+            "blocks": [
+                {"view_id": "classification" if len(labels) == 1 else "choice"},
+                {"view_id": "html", "html_template": HTML_TEMPLATE},
+            ],
+            "show_flag": True,
+            "global_css": CSS_FILE_PATH.read_text(),
+        },
+    }
+
+
+@prodigy.recipe(
+    # fmt: off
+    "textcat.openai.fetch",
+    input_path=("Path to jsonl data to annotate", "positional", None, Path),
+    prompt_path=("Path to jinja2 prompt template", "positional", None, Path),
+    output_path=("Path to save the output", "positional", None, Path),
+    labels=("Labels (comma delimited)", "option", "L", lambda s: s.split(",")),
+    lang=("Language to use for tokenizer.", "option", "l", str),
+    model=("GPT-3 model to use for completion", "option", "m", str),
+    examples_path=("Examples file to help define the task", "option", "e", Path),
+    max_examples=("Max examples to include in prompt", "option", "n", int),
+    batch_size=("Batch size to send to OpenAI API", "option", "b", int),
+    segment=("Split sentences", "flag", "S", bool),
+    exclusive_classes=("Make the classification task exclusive", "flag", "E", bool),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    # fmt: on
+)
+def textcat_openai_fetch(
+    input_path: Path,
+    prompt_path: Path,
+    output_path: Path,
+    labels: List[str] = None,
+    lang: str = "en",
+    model: str = "text-davinci-003",
+    batch_size: int = 10,
+    segment: bool = False,
+    examples_path: Optional[Path] = None,
+    max_examples: int = 2,
+    exclusive_classes: bool = False,
+    verbose: bool = False,
+):
+    api_key, api_org = get_api_credentials(model)
+    examples = read_prompt_examples(examples_path, example_class=TextCatPromptExample)
+    if labels is None:
+        msg.fail("textcat.teach requires at least one --label", exits=1)
+    nlp = spacy.blank(lang)
+
+    if segment:
+        nlp.add_pipe("sentencizer")
+
+    # Create partial render of the template
+    if not exclusive_classes and len(labels) == 1:
+        msg.warn(
+            "Binary classification should always be exclusive. Setting "
+            "`exclusive_classes` parameter to True"
+        )
+        exclusive_classes = True
+
+    template = Template(
+        load_template(prompt_path).render(
+            exclusive_classes=exclusive_classes,
+            labels=labels,
+        )
+    )
+
+    # Create OpenAISuggester with ChatGPT parameters
+    openai = TextCatOpenAISuggester(
+        prompt_template=template,
+        labels=labels,
+        max_examples=max_examples,
+        segment=segment,
+        openai_api_org=api_org,
+        openai_api_key=api_key,
+        openai_model=model,
+        openai_timeout_s=10,
```

Comment:
I wonder if there's any harm in increasing this to 20s. Is there a reason the timeout is 10s?